frontstage/views/sign_in.py

```diff
@@ -31,70 +31,38 @@ def login():
     if request.method == 'POST' and form.validate():
         username = request.form.get('username')
         password = request.form.get('password')
-
-        # TODO Consider moving this to a helper function.
-        # Lets get a token from the OAuth2 server
-        try:
-            token_url = app.config['ONS_TOKEN']
-
-            data = {
-                'grant_type': 'password',
-                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
-                'username': username,
-                'password': password,
-            }
-            headers = {
-                'Accept': 'application/json',
-                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
-            }
-
-            oauth2_response = requests.post(url=token_url, data=data,
-                                            headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                                                   app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
-            # Check to see that this user has not attempted to login too many times or that they have not forgot to
-            # click on the activate account URL in their email by checking the error message back from the OAuth2 server
-            if oauth2_response.status_code == 401:
-                oauth2Error = json.loads(oauth2_response.text)
-                if 'Unauthorized user credentials' in oauth2Error['detail']:
+        sign_in_data = {
+            "username": username,
+            "password": password
+        }
+        response = api_call('POST', app.config['SIGN_IN_URL'], json=sign_in_data)
+
+        # Handle OAuth2 authentication errors
+        if response.status_code == 502:
+            error_json = json.loads(response.text).get('error')
+            if error_json.get('status_code') == 401:
+                error_message = error_json.get('data', {}).get('detail')
+                if 'Unauthorized user credentials' in error_message:
                     return render_template('sign-in/sign-in.html', _theme='default', form=form, data={"error": {"type": "failed"}})
-                elif 'User account locked' in oauth2Error['detail']:
-                    logger.warning('User account is locked on the OAuth2 server')
-                    return render_template('sign-in/sign-in.trouble.html', _theme='default', form=form,
-                                           data={"error": {"type": "account locked"}})
-                elif 'User account not verified' in oauth2Error['detail']:
-                    logger.warning('User account is not verified on the OAuth2 server')
-                    return render_template('sign-in/sign-in.account-not-verified.html', _theme='default', form=form,
-                                           data={"error": {"type": "account not verified"}})
+                elif 'User account locked' in error_message:
```

Comment:
Could this be http 403?

Reply:
We could make this return 403 from the django server? Then pass that through the api

Reply:
Changing this now will introduce breaking changes as there are at least three separate services using the django server

Reply:
Fair point

Reply:
Yeah I don't think we should change it in the django server but as we're introducing the frontstage-api we can do the horrible mapping there and have a clean API which returns the relevant status codes